scripts/deployment/artifactory.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+# To be executed from the root project directory.
+# Obtains R package from remote artifactory and executes local.sh script with package downloaded to temporary directory.
+# Input arguments:
+# -s [required] shiny webapp location where all shiny resources will be deployed
+# -a [required] artifactory root location where R packages are stored along with latest.txt file pointing to the most recent package version
+# -u [optional] user name to be authenticated against artifactory, set to currently logged in system user if not provided
+# -p [optional] R package bundle full name which should be retrieved from artifactory. To be provided when one don't want to deploy most recent package version.
+
+while getopts ":s:a:u:p:x:" opt; do
+  case $opt in
+    s) shiny_webapp_location="$OPTARG"
+    ;;
+    a) artifactory_root="$OPTARG"
+    ;;
+    u) user_name="$OPTARG"
+    ;;
+    p) package_location="$OPTARG"
+    ;;
+    x) password="$OPTARG"
+    ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+    ;;
+  esac
+done
+
+#quitting on error
+set -e
+
+if [ -z "${shiny_webapp_location}" ]; then
+    echo -s required input parameter value with shiny webapp location was not provided!
+    exit 1
+fi
+
+if [ -z "${artifactory_root}" ]; then
+    echo -a required input parameter value with artifactory root location was not provided!
+    exit 1
+fi
+
+if [ -z "${user_name}" ]; then
+    user_name=`whoami`
+    echo setting artifactory user to current user: ${user_name}
+fi
+
+if [ -z "${password}" ]; then
+    password_chunk='--ask-password'
+else
+    password_chunk='--password '$password
+fi
+
+if [ -z "${package_name}" ]; then
```

Comment:
Right, this one diverged from `package_location`. I just renamed it all to `package_file_name` as described in one of the previous comments.